modules/@apostrophecms/admin-bar/ui/apos/components/TheAposAdminBar.vue

```diff
@@ -330,6 +394,35 @@ export default {
     }
   },
   mounted() {
+    // Global event listener that will emit bus events from anywhere via click
+    // To use: Create some UI with the below data attributes
+    // data-apos-bus-event='{"name": "EVENT-NAME", "data": {"FOO": TRUE}}'
+    // Also accepts a simplified name-only string data-apos-bus-event="NAME"
+    // `data` parameter is optional, if present will be passed through the emitted event
+
+    document.body.addEventListener('click', (e) => {
+      if (e.target.getAttribute('data-apos-bus-event')) {
+        const event = e.target.getAttribute('data-apos-bus-event');
+        let name;
+        let json = {};
+        try {
+          json = JSON.parse(event);
+          name = json.name || false;
+        } catch (e) {
+          name = event;
+        }
+        if (name) {
+          apos.bus.$emit(name, json.data || null);
+        } else {
+          console.error('Apostrophe bus events require a name');
+          apos.notify('Something went wrong', { type: 'error' });
+        }
+      }
+    }, false);
```

Comment:
@boutell Global admin front-end JS is typically injected via `asset/index.js`, maybe this should move there?

FWIW that pattern feels pretty unintuitive when wanting to sprinkle in something like this. Is there a better home?

Reply:
Global admin front-end JS belongs in "apps" rather than "components." This code should live in `modules/@apostrophecms/ui/ui/apos/apps/apos-bus-event.js`. Any files found in `ui/apos/apps` get pushed to logged-in users as executable code which is what we need here.

Reply:
Global admin front-end JS belongs in "apps" rather than "components." This code should live in modules/@apostrophecms/ui/ui/apos/apps/apos-bus-event.js. Any files found in ui/apos/apps get pushed to logged-in users as executable code which is what we need here.

Reply:
Oh cool, haven't done this yet .. updated